src/components/OptionsSelector/BaseOptionsSelector.js

```diff
@@ -59,8 +60,14 @@ class BaseOptionsSelector extends Component {
                 if (!focusedOption) {
                     return;
                 }
-
-                this.selectRow(focusedOption);
+                if (!this.state.shouldDisableRowSelection) {
+                    this.setState({shouldDisableRowSelection: true});
+                    let result = this.selectRow(focusedOption);
+                    if (!(result instanceof Promise)) {
+                        result = Promise.resolve();
+                    }
+                    setTimeout(() => result.finally(() => this.setState({shouldDisableRowSelection: false})), 300);
```

Comment:
And I think 300 is little less of a timeout, I tried to do 4 presses continuously 3 where blocked and 1 went through causing same issue. I think 400 or 500 should be good. Kindly test and update the values appropriately.

Reply:
Just to mention don't go to higher numbers than what is required since it might not be a good user experience to block action for more time.